.travis/build.sh

```diff
@@ -43,8 +43,51 @@ cat > pytest.ini << EOF
 addopts=-p no:logging --color=no --force-flaky
 EOF
 
+if [ "$TARGET_HOST" == "arm-linux-gnueabihf" ] || [ "$TARGET_HOST" == "arm-linux-gnueabihf" ]
+then
+    export QEMU_LD_PREFIX=/usr/"$TARGET_HOST"/
+    export MAKE_HOST="$TARGET_HOST"
+    export AR="$TARGET_HOST"-ar
+    export AS="$TARGET_HOST"-as
+    export CC="$TARGET_HOST"-gcc
+    export CXX="$TARGET_HOST"-g++
+    export LD="$TARGET_HOST"-ld
+    export STRIP="$TARGET_HOST"-strip
 
-if [ "$SOURCE_CHECK_ONLY" == "false" ]; then
+    wget -q https://zlib.net/zlib-1.2.11.tar.gz \
+    && tar xvf zlib-1.2.11.tar.gz \
+    && cd zlib-1.2.11 \
+    && ./configure --prefix="$QEMU_LD_PREFIX" \
+    && make \
+    && sudo make install 
+    cd .. && rm zlib-1.2.11.tar.gz && rm -rf zlib-1.2.11
+
+    wget -q https://www.sqlite.org/2018/sqlite-src-3260000.zip \
+    && unzip sqlite-src-3260000.zip \
+    && cd sqlite-src-3260000 \
+    && ./configure --disable-tcl --enable-static --disable-readline --disable-threadsafe --disable-load-extension --host="$TARGET_HOST" --prefix="$QEMU_LD_PREFIX" \
+    && make \
+    && sudo make install 
+    cd .. && rm sqlite-src-3260000.zip && rm -rf sqlite-src-3260000
+
+    wget -q https://gmplib.org/download/gmp/gmp-6.1.2.tar.xz \
+    && tar xvf gmp-6.1.2.tar.xz \
+    && cd gmp-6.1.2 \
+    && ./configure --disable-assembly --prefix="$QEMU_LD_PREFIX" --host="$TARGET_HOST" \
+    && make \
+    && sudo make install 
+    cd .. && rm gmp-6.1.2.tar.xz && rm -rf gmp-6.1.2
```

Comment:
If you checkout the dependencies in an architecture specific directory in the `dependencies` subdirectory we can cache them across builds, speeding things up and saving us a whole lot of extra CPU cycles :wink: